doc/SAI-Packet-Action-Group.md

```diff
@@ -0,0 +1,729 @@
+    SAI_PACKET_ACTION_GROUP_ATTR_TYPE,
+
+    /**
+     * @brief Number of group members in the Packet action group
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_PACKET_ACTION_GROUP_ATTR_PACKET_ACTION_MEMBER_COUNT,
+
+    /**
+     * @brief Packet action group member list
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_PACKET_ACTION_GROUP_MEMBER
+     */
+    SAI_PACKET_ACTION_GROUP_ATTR_PACKET_ACTION_GROUP_MEMBER_LIST,
+
+    /**
+     * @brief Packet action active group member. 
+     *
+     * If not specified or is set as SAI_NULL_OBJECT_ID, the first member acts as the active member in 
+     * the group. 
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @default SAI_NULL_OBJECT_ID
+     * @validonly SAI_PACKET_ACTION_GROUP_ATTR_TYPE ==  SAI_PACKET_ACTION_GROUP_TYPE_FAST_FAILOVER
+     */
+    SAI_PACKET_ACTION_GROUP_ACTIVE_MEMBER, 
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_PACKET_ACTION_GROUP_ATTR_END,
+
+    /** Custom range base value */
+    SAI_PACKET_ACTION_GROUP_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_PACKET_ACTION_GROUP_ATTR_CUSTOM_RANGE_END
+
+} sai_packet_action_group_attr_t;
+
+typedef enum _sai_packet_action_group_member_status
+{
+   /** This member is not forwarding traffic */
+   SAI_PACKET_ACTION_GROUP_MEMBER_INACTIVE,
+
+   /** This member is active and is forwarding traffic */
+   SAI_PACKET_ACTION_GROUP_MEMBER_ACTIVE
+
+} sai_packet_action_group_member_status;
+
+typedef enum _sai_packet_action_group_member_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_START,
+
+    /**
+     * @brief Packet Action Group ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_PACKET_ACTION_GROUP
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_PACKET_ACTION_GROUP_ID = SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_START,
+
+    /**
+     * @brief Packet Action Group Member Type
+     *
+     * @type sai_packet_action_group_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_TYPE,
+
+    /**
+     * @brief Packet Action Group Action List
+     *    
+     * @type sai_packet_action_group_action_list_t pktactionlist
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_ACTION_LIST,
+
+    /**
+     * @brief Packet Action Group Redirect Port/LAG interface
+     *
+     * This is the egress port/LAG interface out of which the packet needs to be
+     * sent out of the member. If this object is set as SAI_NULL_OBJECT_ID or
+     * is not specified. Then the packet needs to be dropped.
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_REDIRECT_INTERFACE,
```

Comment:
How does this impact the other attribute we have which support fast fail over natively